src/rpcwallet.cpp

```diff
@@ -1429,10 +1429,12 @@ Value walletpassphrase(const Array& params, bool fHelp)
             "walletpassphrase <passphrase> <timeout> [stakingonly]\n"
             "Stores the wallet decryption key in memory for <timeout> seconds.\n"
             "if [stakingonly] is true sending functions are disabled.");
-    if (params[1].get_int64() > 99999999 || params[1].get_int64() < 0) {
-    	throw runtime_error(
-            "walletpassphrase <passphrase> <timeout> [stakingonly]\n"
-            "Maximum timeout value is 99999999 use timeout of 0 to never re-lock, negative values not allowed");
+    if(params.size() > 2) {
```

Comment:
should this be >= 2 ??

Reply:
Yes, because element [0] is 1, and [1] is 2.

Reply:
Yes, but should be `>=`. I'll fix it.

Reply:
I couldn't reproduce an error for the second array access. But it's definitely good practice to test it before accessing it. I'll do that